src/index.js

```diff
@@ -3306,47 +3310,45 @@ async function retrieveMetadata() {
 
 	if ( queueItem ) {
 
-		let uri    = queueItem.dataset.file,
-			revoke = false;
+		let uri    = queueItem.dataset.file;
+		let file;
 
 		waitingMetadata++;
 		delete queueItem.dataset.retrieve;
+		let metadata;
 
-		queryMetadata: {
-			if ( queueItem.handle ) {
-				try {
-					if ( await queueItem.handle.requestPermission() != 'granted' )
-						break queryMetadata;
+		if ( queueItem.handle ) {
+			// Fetch metadata from File object
+			if ( await queueItem.handle.requestPermission() !== 'granted' )
+				return;
```

Comment:
`return`ing here doesn't decrement the counter nor makes the recursive call. I agree the "goto" style is not the best practice, so maybe use a helper function?

Reply:
Resolved by eliminating `waitingMetadata` and using `map` to run queued items without metadata in parallel.